src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -106,15 +106,24 @@ private class LatestDiagnosticsForSpanGetter
                 _includeSuppressedDiagnostics = includeSuppressedDiagnostics;
             }
 
-            public async Task<bool> TryGetAsync(List<DiagnosticData> list, CancellationToken cancellationToken)
+            public async Task<bool> TryGetAsync(List<DiagnosticData> list, Func<string, IDisposable> addOperationScopeOpt, CancellationToken cancellationToken)
             {
                 try
                 {
                     var containsFullResult = true;
                     foreach (var stateSet in _stateSets)
                     {
-                        using (RoslynEventSource.LogInformationalBlock(FunctionId.DiagnosticAnalyzerService_GetDiagnosticsForSpanAsync, stateSet.Analyzer, cancellationToken))
+                        IDisposable disposableScope = null;
+                        RoslynEventSource.LogBlock? logBlock = null;
+                        try
                         {
+                            if (addOperationScopeOpt != null)
+                            {
+                                var analyzerTypeName = this._owner.HostAnalyzerManager.GetAnalyzerTypeName(stateSet.Analyzer);
+                                disposableScope = addOperationScopeOpt.Invoke(analyzerTypeName);
+                                logBlock = RoslynEventSource.LogInformationalBlock(FunctionId.DiagnosticAnalyzerService_GetDiagnosticsForSpanAsync, analyzerTypeName, cancellationToken);
```

Comment:
@sharwell Moving this `LogInformationalBlock` within the check `addOperationScopeOpt != null` fixes the issue we had discussed offline where the logging was happening for analyzers on code paths outside of Ctrl + Dot

Reply:
💭 Seems like we still want to know when analyzers are running outside of lightbulb scenarios, just not for cases where the results are already computed and available from a cache.